directory/cli/src/host.py

```diff
@@ -0,0 +1,181 @@
+		ipa_wrapper_command.create(
+			'create',
+			ipa_command='host-add',
+			argument_name='hostname',
+			options=_host_options(),
+			transform_options_callback=_transform_create_options,
+			handle_result_callback=_handle_create_result,
+			help='Create new host'
+		),
+		ipa_wrapper_command.create(
+			'modify',
+			ipa_command='host-mod',
+			argument_name='hostname',
+			options=_host_options(),
+			transform_options_callback=_transform_modify_options,
+			handle_result_callback=_handle_modify_result,
+			help='Modify existing host'
+		),
+		ipa_wrapper_command.create(
+			'delete',
+			ipa_command='host-del',
+			argument_name='hostname',
+			handle_result_callback=_handle_delete_result,
+			help='Delete existing host'
+        )]
+	
+	for command in wrapper_commands:
+		host.add_command(command)
+
+def _additional_data_for_list():
+	return{
+		'ip-address': _get_ip(),
+		'hostgroups': _hostgroups_by_name()
+	}
+
+def _get_ip():
+	host_ips_by_name = {}
+	for host_data in _all_hosts():
+		try:	
+			command = 'dnsrecord-find'
+			host_name = host_data['Host name'][0]
+			host_name_text = re.sub(r'\..*$',"",host_name)
+			#print("*********HOSTNAME IS EQUAL TO:" + host_name_text + "*************")
+			domain_name= re.sub(r'^.*?\.',"",host_name)
+			#print("*********PRIMARY DOMAIN IS EQUAL TO:"+ domain_name  + "**************\n")
+			host_name_instruct = '--name='+host_name_text
+			args = [domain_name, host_name_instruct]
+
+			ip_result = ipa_utils.ipa_run(command, args, record=False)
+			ip_result_parsed = ipa_utils.parse_find_output(ip_result)
+			
+			ip_address_dict = ip_result_parsed[0]
+			ip_address_list = ip_address_dict["A record"]
+			#print( ip_address_list)
+			host_ips_by_name[host_name] = ip_address_list
+		except KeyError:
+			continue	
+	return host_ips_by_name
+	
+def _hostgroups_by_name():
+	hostgroups_by_name = {}
+	for hostgroup_data in _all_hostgroups():
+		try:
+			host_group_name = hostgroup_data['Host-group'][0]
+			hostgroups_by_name[host_group_name] = hostgroup_data
+		except KeyError:
+			continue
+	return hostgroups_by_name	
+
+def _all_hosts():
+	public_hosts = ipa_utils.ipa_find('host-find')
+	#private_hosts = ipa_utils.ipa_find('host-find', ['--private'])
+	#print("finished all_hosts\n")
+	return public_hosts# + private_hosts
+	
+
+def _all_hostgroups():
+	public_hostgroups = ipa_utils.ipa_find('hostgroup-find')
+	return public_hostgroups
+
+def _host_options():
+	return {
+		'--password': {'help': 'Host password'},
+		'--ip-address': {'help': 'IP address of host'},
+	}
+
+def _transform_create_options(argument, options):
+	_validate_blacklist_hosts(argument)
+	return OptionTransformer(argument, options).\
+		options
+
+def _validate_blacklist_hosts(argument, options={}):
+	if argument in HOST_BLACKLIST:
+		error = "The host " + argument + " is a restricted host"
+		raise click.ClickException(error)
+
+def _transform_modify_options(argument, options):
+	return OptionTransformer(argument, options).\
+		options
```

Comment:
Here, and in the other places we now use `OptionTransformer` in this file and `hostgroup.py`, I think we can simplify this. `OptionTransformer` doesn't change the options at all by default, just provides a convenient wrapper for performing various transforms on the options to be passed to `ipa`, so I believe in each case this is effectively identical to just `return options`. This in turn is identical to what the default argument does:

https://github.com/alces-software/userware/blob/be60b773e62c6700fc87587d9dde9e5d977ea4cc/directory/cli/src/ipa_wrapper_command.py#L16

so in the cases where we're not doing anything else in these `transform_options_callback` functions (like this one) I believe we can just remove the function and argument entirely. Unless of course we do want to call `_validate_blacklist_hosts(argument)` in this function as in `_transform_create_options` (don't know enough about what we want to do here to know if that makes sense), in which case we could share a function for both these callbacks as in `hostgroup.py`.